### PR TITLE
Kgp 500 announment

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
image

### CODE REVIEW

File: src/components/Announcement/AnnoucementView.tsx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    css`
+      background-color: ${({ theme }) => theme.tabActive};
+      color: ${({ theme }) => theme.text};
+    `};
+`
+
+const Title = styled.div`
+  font-weight: 500;
+  font-size: 20px;
+  font-weight: 500;
```
Comment (XiaoYhun) [inline src/components/Announcement/AnnoucementView.tsx:72]: duplicated

File: src/services/announcement.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      query: params => ({
+        url: `/v1/messages/announcements`,
+        params,
+      }),
+      transformResponse,
+    }),
+    getPrivateAnnouncements: builder.query<Response, Params>({
+      query: ({ account, ...params }) => ({
+        url: `/v1/users/${account}/notifications`,
+        params: { ...params, excludedTemplateIds: '2,29' }, // todo danh config env
```
Comment (namgold) [inline src/services/announcement.ts:43]: todo

File: src/services/announcement.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      transformResponse,
+    }),
+    ackPrivateAnnouncements: builder.mutation<
+      Response,
+      { account: string; action: 'read' | 'clear-all' | 'read-all'; ids?: number[] }
+    >({
+      query: ({ account, action, ids }) => {
+        const body: { excludedTemplateIds?: number[]; ids?: number[] } = { ids }
+        if (action === 'read-all' || action === 'clear-all') {
+          body.excludedTemplateIds = [2, 29] // todo danh config env
```
Comment (namgold) [inline src/services/announcement.ts:54]: todo

**Review by XiaoYhun — DISMISSED** [review]

File: src/components/Announcement/Popups/SimplePopup.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    [NotificationType.ERROR]: <IconFailure color={color} />,
+  }
+
+  const navigate = useNavigate()
+  const onClickLink = () => {
+    link && navigate(link)
+  }
+  return (
+    <Box>
+      <RowNoFlex>
```
Comment (XiaoYhun) [inline src/components/Announcement/Popups/SimplePopup.tsx:51]: I don't get this component, AutoRow added flex-wrap, and you remove it from AutoRow? Why not use Row in first place?

Comment (nguyenhoaidanh) [reply src/components/Announcement/Popups/SimplePopup.tsx:51]: thinking now... old code. will update

File: src/components/Announcement/AnnoucementItem.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  justify-content: space-between;
+  overflow: hidden;
+  height: ${HEIGHT};
+`
+
+const Image = styled.img`
+  width: ${HEIGHT};
+  max-height: ${HEIGHT};
+  border-radius: 8px;
+  object-fit: contain;
```
Comment (XiaoYhun) [inline src/components/Announcement/AnnoucementItem.tsx:74]: consider to use object-fit: scale-down, I see some images that lower than 100px is magnified, make it losing detail
image

Comment (XiaoYhun) [reply src/components/Announcement/AnnoucementItem.tsx:74]: also I see a bug of the TopBanner 
image

Comment (XiaoYhun) [reply src/components/Announcement/AnnoucementItem.tsx:74]: image
icons not centered, See more position looks weird

Comment (nguyenhoaidanh) [reply src/components/Announcement/AnnoucementItem.tsx:74]: - arrow shape: this is behavior of `swiper`
- bell icon image from design
- See more see less/ see less: I will give feedback to the designer.

Comment (XiaoYhun) [reply src/components/Announcement/AnnoucementItem.tsx]: there should be someway to customize the arrows, it look really bad for production, also there is bug of border-radius while swiping

File: src/components/Announcement/AnnoucementItem.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  -webkit-line-clamp: 2;
+  -webkit-box-orient: vertical;
+  overflow: hidden;
+  text-overflow: ellipsis;
+
+  > p {
+    margin: 0;
+  }
+`
+const Time = styled.span`
```
Comment (nguyenhuudungz) [inline src/components/Announcement/AnnoucementItem.tsx:53]: enter

File: .env [hunk 7]
```diff
@@ -39,3 +33,4 @@ REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.dev.kyberengineering.io/read-
 
 # Kyber DAO Apis
 REACT_APP_KYBER_DAO_STATS_API=https://kyberswap-dao-stats.kyberengineering.io
+REACT_APP_NOTIFICATION_IGNORE_TEMPLATE_IDS=2,29
```
Comment (nguyenhuudungz) [inline .env:36]: add todo pls

File: .env.dev [hunk 8]
```diff
@@ -37,4 +31,5 @@ REACT_APP_LIMIT_ORDER_API_WRITE=https://limit-order.dev.kyberengineering.io/writ
 REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.dev.kyberengineering.io/read-ks/api
 
 # Kyber DAO Apis
-REACT_APP_KYBER_DAO_STATS_API=https://kyberswap-dao-stats.dev.kyberengineering.io
\ No newline at end of file
+REACT_APP_KYBER_DAO_STATS_API=https://kyberswap-dao-stats.dev.kyberengineering.io
+REACT_APP_NOTIFICATION_IGNORE_TEMPLATE_IDS=2,29
```
Comment (nguyenhuudungz) [inline .env.dev:35]: add todo pls

File: .env.production [hunk 9]
```diff
@@ -37,4 +31,5 @@ REACT_APP_LIMIT_ORDER_API_WRITE=https://limit-order.kyberswap.com/write/api
 REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.kyberswap.com/read-ks/api
 
 # Kyber DAO Apis
-REACT_APP_KYBER_DAO_STATS_API=https://kyberswap-dao-stats.kyberengineering.io
\ No newline at end of file
+REACT_APP_KYBER_DAO_STATS_API=https://kyberswap-dao-stats.kyberengineering.io
+REACT_APP_NOTIFICATION_IGNORE_TEMPLATE_IDS=2,11
```
Comment (nguyenhuudungz) [inline .env.production:35]: add todo pls

File: .env.stg [hunk 10]
```diff
@@ -40,3 +34,4 @@ REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.stg.kyberengineering.io/read-
 
 # Kyber DAO Apis
 REACT_APP_KYBER_DAO_STATS_API=https://kyberswap-dao-stats.stg.kyberengineering.io
+REACT_APP_NOTIFICATION_IGNORE_TEMPLATE_IDS=2,11
```
Comment (nguyenhuudungz) [inline .env.stg:37]: add todo pls

File: src/components/Announcement/MoreAction.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+`
+const customStyle = css`
+  padding: 16px;
+  top: 3rem;
+`
+const MenuMoreAction = ({ showClearAll, clearAll }: { clearAll: () => void; showClearAll: boolean }) => {
+  return (
+    <MenuFlyout modalWhenMobile={false} trigger={<MoreHorizontal cursor="pointer" />} customStyle={customStyle}>
+      <Column gap="16px">
+        {/* <MenuItem onClick={showNotificationModal}>
```
Comment (nguyenhuudungz) [inline src/components/Announcement/MoreAction.tsx:28]: why not remove it

File: src/components/Announcement/Popups/CtaButton.tsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+import { ButtonEmpty, ButtonPrimary } from 'components/Button'
+import useTheme from 'hooks/useTheme'
+
+function CtaButton({
+  data,
+  className = '',
+  color,
+  onClick,
+}: {
+  data: AnnouncementCTA
```
Comment (nguyenhuudungz) [inline src/components/Announcement/Popups/CtaButton.tsx:14]: what is cta

Comment (nguyenhoaidanh) [reply src/components/Announcement/Popups/CtaButton.tsx:14]: Call to Action

File: .env.production (hunk 9 above)
Comment (nguyenhoaidanh) [reply .env.production:37]: added

File: src/components/Announcement/Popups/TopBanner.tsx [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+const Content = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: 4px;
+  gap: 8px;
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    font-size: 14px;
+    flex: 1;
+    width:100%;
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply src/components/Announcement/Popups/TopBanner.tsx:46]: khó thế mà cũng nhìn ra

**Review by nguyenhuudungz — APPROVED** [review]

**Pull request merged**